components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +384,161 @@ public void setAlias(String alias) {
+            }
+            certificateInfo.setCertValue(jsonCertificateInfoObject.getString(CERT_VALUE));
+            certificateInfos.add(certificateInfo);
+        }
+        return certificateInfos.toArray(new CertificateInfo[lengthOfJsonArray]);
+    }
+
+    /**
+     * handle the certificate which is in encoded format.
+     *
+     * @param certificateValue
+     * @return array of certificate value and thumbPrint of each certificates.
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] handleEncodedCertificate(String certificateValue) throws NoSuchAlgorithmException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling encoded certificates: " + certificateValue);
+        }
+        String decodedCertificate = new String(Base64.getDecoder().decode(certificateValue));
+        return createCertificateInfo(decodedCertificate);
+    }
+
+    /**
+     * handle the certificate which is in plain text format.
+     *
+     * @param certificateValue
+     * @return array of certificate value and thumbPrint of each certificates.
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] handlePlainTextCertificate(String certificateValue) throws NoSuchAlgorithmException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling plain text certificate: " + certificateValue);
+        }
+        return createCertificateInfo(certificateValue);
+    }
+
+    /**
+     * create certificate info by assigning certificate value and thumbPrint value.
+     * @param decodedCertificate
+     * @return
+     * @throws NoSuchAlgorithmException
+     */
+    private CertificateInfo[] createCertificateInfo(String decodedCertificate) throws
+            NoSuchAlgorithmException {
+
+        int numberOfCertificates = StringUtils.countMatches(decodedCertificate, IdentityUtil.PEM_BEGIN_CERTFICATE);
+        if (numberOfCertificates > 1) {
+            if (log.isDebugEnabled()) {
+                log.debug(numberOfCertificates + " certificates have been found. ");
+            }
+        }
+        List<CertificateInfo> certificateInfoArrayList = new ArrayList<>();
+        for (int ordinal = 1; ordinal <= numberOfCertificates; ordinal++) {
+            String certificateVal;
+            certificateVal = Base64.getEncoder().encodeToString(IdentityApplicationManagementUtil.extractCertificate
+                    (decodedCertificate, ordinal).getBytes(StandardCharsets.UTF_8));
+            CertificateInfo certificateInfo = new CertificateInfo();
+            String thumbPrint = IdentityApplicationManagementUtil.generateThumbPrint(certificateVal);
+            if (log.isDebugEnabled()) {
+                log.debug("ThumbPrint of the certificate is: " + thumbPrint);
+            }
+            certificateInfo.setThumbPrint(thumbPrint);
+            certificateInfo.setCertValue(certificateVal);
```

Comment:
Extract to a method which returns `CertificateInfo` given the `certificateVal`

Reply:
After resolving the other PR comments the above logic is used only in one place. Is it necessary to add to a separate method?